modules/core/include/visp3/core/vpImageTools.h

```diff
@@ -409,6 +420,168 @@ void vpImageTools::crop(const unsigned char *bitmap, unsigned int width, unsigne
+  float min_1 = std::min(transcp[0][0], std::min(transcp[0][1], std::min(transcp[0][2], std::min(transcp[0][3], transcp[0][4])))); ///< min(transcp(0,:)) 
+
+  float max_2 = std::max(transcp[1][0], std::max(transcp[1][1], std::max(transcp[1][2], std::max(transcp[1][3], transcp[1][4])))); ///< max(transcp(1,:)) 
+  float min_2 = std::min(transcp[1][0], std::min(transcp[1][1], std::min(transcp[1][2], std::min(transcp[1][3], transcp[1][4])))); ///< min(transcp(1,:)) 
+
+  vpMatrix imgy(floor(max_1 - min_1 + 1), floor(max_2 - min_2 + 1));
+  vpMatrix imgx(floor(max_2 - min_2 + 1), floor(max_1 - min_1 + 1));
+
+  // meshgrid(min_2:max_2,min_1:min_1)
+  for (unsigned int i = 0; i < imgy.getRows(); i++) {
+    for (unsigned int j = 0; j < imgy.getCols(); j++) {
+      imgy[i][j] = min_2 + j;
+    }
+  }
+
+  // meshgrid(min_1:max_1,min_2:min_2)
+  for (unsigned int i = 0; i < imgx.getRows(); i++) {
+    for (unsigned int j = 0; j < imgx.getCols(); j++) {
+      imgx[i][j] = min_1 + j;
+    }
+  }
+
+  imgx = imgx.t();
+
+  //Backwards Transform (inverse mapping)
+
+  // Initialize Inverse Mapping matrix
+  vpMatrix X(3, imgx.getRows()*imgx.getCols());
+
+  vpColVector B(3);
+
+  // X = [ imgx(:) , imgy(:) , ones(imgx.getRows() * imgx.getCols(), 1) ]'
+  for (unsigned int i = 0; i < X.getRows(); i++) {
+    for (unsigned int j = 0; j < X.getCols(); j++) {
+      if (i == 0)  X[i][j] = imgx[j % imgx.getRows()][j / imgx.getRows()];
+      if (i == 1)  X[i][j] = imgy[j % imgy.getRows()][j / imgy.getRows()];
+      if (i == 2)  X[i][j] = 1;
+    }
+  }
+
+  // perform T \ X
+  for (unsigned int j = 0; j < X.getCols(); j++) {
+    B[0] = X[0][j];
+    B[1] = X[1][j];
+    B[2] = X[2][j];
+    T.solveBySVD(B, B); // Converted Columns of X to a ColVector for Linear Solver
+    X[0][j] = B[0];
+    X[1][j] = B[1];
+    X[2][j] = B[2];
+  }
+
+  // reshape row 1 and row 2 of X into imgx and imgy respectively
+  for (unsigned int j =0; j< imgx.getCols(); j++) {
+    for (unsigned int i =0; i< imgx.getRows(); i++) {
+      imgx[i][j]= X[0][j*imgx.getRows()+i];      
+      imgy[i][j]= X[1][j*imgx.getRows()+i];      
+    }
+  }
+
+  imgx = imgx.t();
+  imgy = imgy.t();
+
+  Iwarp.resize(imgx.getRows(),imgx.getCols());
+
+  // Use nearest neighbour interpolation
+  for (unsigned int i = 0; i < Iwarp.getRows(); i++) {
+    for (unsigned int j = 0; j < Iwarp.getCols(); j++) {
+      Iwarp[i][j] = getPixelClampedWarp(I, imgx[i][j], imgy[i][j]);
+    }
+  }
+
+  if (T[2][0] != 0 || T[2][1] != 0) {
+    vpImage<Type> Iwarp_temp = Iwarp;
+    signed int iSign = -1 * ((T[2][1] > 0) - (T[2][1] < 0)), jSign = -1 * ((T[2][0] > 0) - (T[2][0] < 0));
+
+    for (unsigned int i =  0; i < Iwarp.getRows(); i++) {
+      for (unsigned int j = 0; j < Iwarp.getCols(); j++) {
+
+        if (floor((i - T[2][1]) / Iwarp.getRows()) == 0 && floor((j - T[2][0]) / Iwarp.getCols()) == 0)
+          Iwarp[i][j] =  Iwarp_temp[i - (int)T[2][1]][j - (int)T[2][0]];
```

Comment:
`floor((i - T[2][1]) / Iwarp.getRows()` should be computed just once for each row between line 566 and 567